### ISSUE TITLE
Confusing UI: What do the "2" and "1" buttons mean?

### ISSUE THREAD

**Issue by shawnmjones** [oduwsdl/tmvis#62]
One particular part of the UI is confusing.

1. I have entered the URI-M https://wayback.archive-it.org/2950/20120110014403/http://OccupyHouston.org/es and clicked "Calculate # of Thumbnails"
2. I then clicked "Calculate Unique"
3. I am confronted with instructions that state "The options shown below are based on the amount of difference between the mementos." and am presented with 3 buttons as shown in the screenshot: "2", "1", and "Generate Thumbnails".

The instructions in step number 3 imply that "2" and "1" provide options. After some consideration, I assume that they are paginating something, but I've clicked on both and see no change. Clicking "1" changes the message beneath the buttons to "(estimated 1 minutes to generate)". Clicking "2" changes the message beneath to "(estimated 2 minutes to generate)". What do clicking them change? Do these buttons alter the amount of time dedicated to generating thumbnails?

My suggestion is to be more explicit in the instructions about what is expected of the user.

![image](https://user-images.githubusercontent.com/4702031/64350027-33681480-cfb5-11e9-96d3-f6bbe0e35c4b.png)

**Comment by machawk1** [comment]
@shawnmjones It looks like @a-mabe addressed this in 3b48ca8 but the issue remains open. For reference, here is the current UI:

![Screen Shot 2020-06-18 at 10 39 08 AM](https://user-images.githubusercontent.com/2514780/85034421-17062480-b150-11ea-92ea-fb66a50132ef.png)

I think it is closer to meeting your requirement with the additional instructions but wanted your feedback. I personally feel that the numerical radio need an associated label for context outside of the paragraph description, e.g.,

**THUMBNAILS TO GENERATE**
[ 14 ] [ 8 ] [ 7 ] [ 5 ] [ 4 ] [ Generate ]

@ibnesayeed Also has an eye for UI/UX, so further feedback welcomed from him as well.

**Comment by ibnesayeed** [comment]
I agree with @shawnmjones that the UI element used can be confusing, especially if numbers come out to be sequential, one might think these are paginations of some sort. I would go with @machawk1's suggestion of adding a label there. However, I would change the UI element as well. The right UI element for this purpose is **radio buttons** (one may use select dropdown, but that would be less accessible than radio buttons for smaller list of items). I would go with the [inline radio buttons arrangement](https://getbootstrap.com/docs/4.0/components/forms/#inline) (not check boxes!).

How about something like the following:

```
Number of Thumbnails:    ○ 14    ⏺ 8    ○ 7    ○ 5    ○ 4
                         [Generate Thumbnails]
```

**Comment by machawk1** [comment]
Good points with regard with accessibility, @ibnesayeed. I like your solution but have not dug into the code recently to the point where I can determine if it could result in a large number of radios (TODO).

**Comment by shawnmjones** [comment]
With the recent improvements to the UI, I now understand what the numbers mean. My original issue has been addressed. With respect to the point raised by @ibnesayeed on radio buttons, I'd like to echo @machawk1's concerns about the large number of radios. 

Would a drop-down be better? The drop-down may grow to a very large size, but it will take up less space initially.

What about a free-form text input that allows the user to insert any number and some JavaScript that makes sure the number supplied is acceptable (e.g., in range)?

**Comment by ibnesayeed** [comment]
> Would a drop-down be better? The drop-down may grow to a very large size, but it will take up less space initially.

If the number of proposed items are something we can count on our fingers then radio buttons save an extra click and have all the options visible without any efforts, but when items grow more than a handful then dropdown is the way to go.

> What about a free-form text input that allows the user to insert any number and some JavaScript that makes sure the number supplied is acceptable (e.g., in range)?

I think those numbers are proposed by the algorithm as potential good breakpoints, so a free-form entry will not be a good idea.

**Comment by machawk1** [comment]
The upper bound of number of options is not apparent without more insights into the code. Perhaps @dhruv282 or @a-mabe can provide some info on the upper bound of options here.

Radios or a dropdown are good for a few options but dropdowns, too, have their limits. In [Mink](https://github.com/machawk1/mink), above 100-ish mementos, I move from dropdown to [Miller columns](https://en.wikipedia.org/wiki/Miller_columns), albeit, it's with temporal data, which lends itself to sorting this day, unlike a list of numbers (here).

Given they are a predefined set that are not simply enumeration, freeform entry might not work as expected.

**Comment by ibnesayeed** [comment]
> Radios or a dropdown are good for a few options but dropdowns, too, have their limits. In [Mink](https://github.com/machawk1/mink), above 100-ish mementos, I move from dropdown to [Miller columns](https://en.wikipedia.org/wiki/Miller_columns), albeit, it's with temporal data, which lends itself to sorting this day, unlike a list of numbers (here).

Miller columns are applicable to hierarchical data. Another common UI that was being used for large lists was called combo-box. It was not a native element, but various JS UI libraries introduced it. It is a combination of free-form text input with the appearance of a dropdown list. One can either select an item from the list, or simply type something to filter the list smaller (something like search suggestions). The use case was so common that HTML5 introduced something called a datalist, which is a native way of suggesting potential options to an input field. This does not prevent people from entering data in the input field that is not in the datalist though.

That said, in this particular case if the upper bound is small enough then radio buttons will be a good option otherwise dropdowns.

**Comment by dhruv282** [comment]
The upper bound for the number of options is 10 but there could be only one option in some cases. I believe this is why Maheedhar implemented these radio buttons initially. 

I think @machawk1's suggestion below of adding a label would be beneficial.
> **THUMBNAILS TO GENERATE**
> [ 14 ] [ 8 ] [ 7 ] [ 5 ] [ 4 ] [ Generate ]

**Comment by ibnesayeed** [comment]
If max options are ten, but most of the time it is below five or six then I would go with the radio button UI. However, I think a dropdown list will be much more consistent when it comes to screen real estate. Situations when there will be only one option given, I would perhaps hide the UI of selection, but if for the consistency sake we want to keep the element in the UI then I think a dropdown would be the least confusing option.
